charts/access/email/templates/deployment.yaml

```diff
@@ -48,6 +48,16 @@ spec:
             - name: auth-id
               mountPath: /var/lib/teleport/plugins/email/auth_id
               subPath: {{ .Values.teleport.identitySecretPath }}
+            {{- if or .Values.mailgun.enabled .Values.smtp.enabled }}
+            - name: {{ .Values.secretVolumeName }}
+              {{- if .Values.mailgun.enabled }}
+              mountPath: "/var/lib/teleport/plugins/email/mailgun_private_key"
+              subPath: "{{ .Values.mailgun.privateKeySecretPath }}"
+              {{- else }}
```

Comment:
Somewhat of a nit, but from a logical pov, I prefer for readability and future-proofing structure like

```
if A and B:
  [...]
  if A
    [...]
  endif
  if B
    [...]
  endif
endif
```